### PR TITLE
support visual studio compilers (2013 & 2014) on windows

### PR DESCRIPTION
**Pull request by kirkshoop**
If you like, login to http://appveyor.com to add windows CI for the json project. the configuration is in appveyor.yml

### CODE REVIEW

File: test/json_unit.cc [hunk 1]
```diff
@@ -150,7 +165,7 @@ TEST_CASE("array")
         json nonarray = 1;
         CHECK_THROWS_AS(nonarray.at(0), std::domain_error);
         CHECK_THROWS_AS(const int i = nonarray[0], std::domain_error);
-        CHECK_NOTHROW(j[21]);
+		SKIP_FOR_VS(CHECK_NOTHROW(j[21]));
```
Comment (Teemperor) [inline test/json_unit.cc]: replace tabs with spaces

File: test/json_unit.cc [hunk 2]
```diff
@@ -162,7 +177,7 @@ TEST_CASE("array")
         const json j_const = j;
         CHECK_THROWS_AS(nonarray_const.at(0), std::domain_error);
         CHECK_THROWS_AS(const int i = nonarray_const[0], std::domain_error);
-        CHECK_NOTHROW(j_const[21]);
+		SKIP_FOR_VS(CHECK_NOTHROW(j_const[21]));
```
Comment (Teemperor) [inline test/json_unit.cc]: replace tabs with spaces

File: test/json_unit.cc [hunk 3]
```diff
@@ -175,11 +190,11 @@ TEST_CASE("array")
         }
 
         const json k = j;
-        CHECK_NOTHROW(k[21]);
+		SKIP_FOR_VS(CHECK_NOTHROW(k[21]));
```
Comment (Teemperor) [inline test/json_unit.cc]: replace tabs with spaces

File: src/json.h [hunk 4]
```diff
@@ -18,6 +18,33 @@
...
+#define u8
+#undef JSON_USE_LITERALS
+#endif
+
+#undef JSON_NO_RETURN
+#define JSON_NO_RETURN __declspec(noreturn)
+
+#define or ||
+#define and &&
+#define not !
```
Comment (Teemperor) [inline src/json.h]: Why not `#include <ciso464>` which contains all defines?

http://www.cplusplus.com/reference/ciso646/

Comment (kirkshoop) [inline src/json.h]: I tried and VS 2013 & 2014 do not appear to implement that header.

**Comment by kirkshoop** [comment]
.editorconfig has been added to ensure that whitespace is maintained automatically.

**Comment by kirkshoop** [comment]
is there something more that I need to address?

**Comment by nlohmann** [comment]
Hi @kirkshoop, sorry for not answering earlier. Though I like the idea of having a Windows version, I am currently working on new code that uses even more C++11 features. When I look at your code, I see that you mostly use the preprocessor to exclude those parts that are not yet supported by Visual Studio. I think I would rather wait until Visual Studio has true C++11 support rather than to fragmentize the project like this - in the end, I am using this project to learn the new C++11 features.

I'm going to close the pull request here, but I would be happy if you would host a Windows-compatible version yourself. I would link to it in the README.

**Pull request closed without merging**